### PR TITLE
[New Template] state/documentdb - DocumentDB cluster

### PR DESCRIPTION
**Pull request by michaelwittig**

### CODE REVIEW

File: state/documentdb.yaml [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+      FromPort: 27017
+      ToPort: 27017
+      SourceSecurityGroupId: {'Fn::ImportValue': !Sub '${ParentBastionStack}-SecurityGroup'}
+  Cluster: # FIXME add monitoring
+    # TODO not yet supported by CloudFormation DeletionPolicy: Snapshot
+    # TODO not yet supported by CloudFormation UpdateReplacePolicy: Snapshot
+    Type: 'AWS::DocDB::DBCluster'
+    Properties:
+      AvailabilityZones:
+      - {'Fn::ImportValue': !Sub '${ParentVPCStack}-AZA'}
```
Comment (andreaswittig) [inline state/documentdb.yaml:107]: What's the impact of deploying to two AZs only?

File: state/documentdb.yaml [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    Description: 'Optional Stack name of parent KMS key stack based on security/kms-key.yaml template (ignored when DBSnapshotIdentifier is set, value used from snapshot).'
+    Type: String
+    Default: ''
+  BackupRetentionPeriod:
+    Description: 'The number of days for which automated backups are retained.'
+    Type: Number
+    MinValue: 1
+    MaxValue: 35
+    Default: 30
+  DBClusterParameterGroupName:
```
Comment (andreaswittig) [inline state/documentdb.yaml:39]: What about adding a `AWS::DocDB::DBClusterParameterGroup` resource as well? Might be interesting to be able to modify the `change_stream_log_retention_duration`, for example. (see https://docs.aws.amazon.com/documentdb/latest/developerguide/cluster_parameter_groups-list_of_parameters.html)

File: state/documentdb.yaml [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+      BackupRetentionPeriod: !Ref BackupRetentionPeriod
+      DBClusterParameterGroupName: !If [HasDBClusterParameterGroupName, !Ref DBClusterParameterGroupName, !Ref 'AWS::NoValue']
+      DBSubnetGroupName: !Ref DBSubnetGroup
+      EnableCloudwatchLogsExports:
+      - audit
+      EngineVersion: '3.6.0'
+      KmsKeyId: !If [HasParentKmsKeyStack, {'Fn::ImportValue': !Sub '${ParentKmsKeyStack}-KeyId'}, !Ref 'AWS::NoValue']
+      MasterUsername: !Ref MasterUsername
+      MasterUserPassword: !Ref MasterUserPassword
+      Port: 27017
```
Comment (grizzly-monkey) [inline state/documentdb.yaml:118]: may be parametrize ?

File: state/documentdb.yaml [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    Properties:
+      AvailabilityZones:
+      - {'Fn::ImportValue': !Sub '${ParentVPCStack}-AZA'}
+      - {'Fn::ImportValue': !Sub '${ParentVPCStack}-AZB'}
+      BackupRetentionPeriod: !Ref BackupRetentionPeriod
+      DBClusterParameterGroupName: !If [HasDBClusterParameterGroupName, !Ref DBClusterParameterGroupName, !Ref 'AWS::NoValue']
+      DBSubnetGroupName: !Ref DBSubnetGroup
+      EnableCloudwatchLogsExports:
+      - audit
+      EngineVersion: '3.6.0'
```
Comment (grizzly-monkey) [inline state/documentdb.yaml:114]: future version will change having a parameter with `3.6.0` as default

File: state/documentdb.yaml (hunk 1 above)
Comment (michaelwittig) [reply state/documentdb.yaml]: I improved the template to use all available AZs

File: state/documentdb.yaml (hunk 2 above)
Comment (michaelwittig) [reply state/documentdb.yaml:65]: done :)

File: state/documentdb.yaml (hunk 3 above)
Comment (michaelwittig) [reply state/documentdb.yaml:162]: done :)

**Pull request merged**